### PR TITLE
Added Function 'HasToken'

### PR DESCRIPTION
**Pull request by gsantosdasilva**

### CODE REVIEW

**Comment by gsantosdasilva** [comment]
What is the reason for the error?

File: src/WebDocs/wwwroot/app/functions/HasToken/samples/001/content.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <script src="/drapo.js"></script>
+    <title>Auhentication</title>
+</head>
+<body>
+    <div>
+        <div>
+            <div d-dataKey="keyValue" d-dataType="object" d-dataProperty-key-name="Key" d-dataProperty-key-value="admin" d-dataProperty-value-name="Value" d-dataProperty-value-value="password" d-dataUrlSet="/Authentication/Login"></div>
```
Comment (silvath) [inline src/WebDocs/wwwroot/app/functions/HasToken/samples/001/content.html:10]: the API Authentication/Login does not exist

File: src/WebDocs/wwwroot/app/functions/HasToken/parameters.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[
```
[the remark this answers is not in the document]
Comment (gsantosdasilva) [reply src/WebDocs/wwwroot/app/functions/HasToken/parameters.json:1]: We don't have parameters in this function, should I keep the '.json' file empty or delete it?

File: src/WebDocs/Controllers/SampleController.cs [hunk 3]
```diff
@@ -127,5 +127,5 @@ public List<string> GetArray(int start = 0, int length = 10, string prefix = nul
             }
             return (array);
         }
-    }
-}
+
```
Comment (silvath) [inline src/WebDocs/Controllers/SampleController.cs:130]: remove this changes

File: src/WebDocs/wwwroot/app/functions/HasToken/samples/001/content.html (hunk 1 above)
Comment (silvath) [reply src/WebDocs/wwwroot/app/functions/HasToken/samples/001/content.html:10]: This api still doesnt exist. we are getting a 404 in console

File: src/WebDocs/wwwroot/app/functions/HasToken/samples/001/content.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            <div d-dataKey="tokenHasvalue" d-dataType="value"></div>
+            <div d-dataKey="functionToken" d-dataType="value" d-dataLoadType="startup" d-dataValue="IF(HasToken(),UpdateDataField(tokenHasvalue,,Yes),UpdateDataField(tokenHasvalue,,No))"></div>
+            <div d-dataKey="functionStartup" d-dataType="function" d-dataLoadType="startup" d-dataValue="{{functionToken}}"></div>
+        </div>
+        <div>
+            <span d-model="{{tokenHasvalue}}"></span>
+            <br />
+            <input type="button" value="Update Token" d-on-click="UpdateToken({{response}}); Execute({{functionToken}})">
+            <input type="button" value="Clear Token" d-on-click="ClearToken(); Execute({{functionToken}})">
+            <input type="button" value="Has Token" d-on-click="HasToken(); Execute({{functionToken}})">
```
Comment (silvath) [inline src/WebDocs/wwwroot/app/functions/HasToken/samples/001/content.html:20]: HasToken is a function that may be used inside an IF

**Pull request closed without merging**